streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedSegmentedBytesStore.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimestampedSegmentedBytesStore extends AbstractSegmentedBytesStore<TimestampedSegment> {
+
+    RocksDBTimestampedSegmentedBytesStore(final String name,
+                                          final String metricScope,
+                                          final long retention,
+                                          final long segmentInterval,
+                                          final KeySchema keySchema) {
+        super(name, metricScope, keySchema, new TimestampedSegments(name, retention, segmentInterval));
```

Comment:
Instead of declaring two subclasses that only function to pass different arguments to the constructor of their superclass, can we just inline these constructors and pass the desired different arguments at the call site?

Reply:
We could. I prefer the current way because it seems easier to use.

Reply:
Will add `RocksDBTimestampedSegmentedBytesStoreTest`.